Upstatement/firestore-jest-mock
# Complete basic TypeScript support

# Description

This branch adds complete TypeScript support for arbitrary imports. Typings are now available package-wide.

If any of these top-level types are awkwardly placed or named improperly, please let me know.

I'd like to see the rest of the code ported to TypeScript eventually, but this will at least help users to use the mock functions correctly. A TypeScript port would likely involve reorganizing some things if it's to work properly.

## How to test

1. Install this branch in a new TypeScript project.
2. Check that your editor consumes the mock function types correctly.
3. Check that special mock database structures produce compiler errors when mistyped (e.g. providing `_collections` as an array produces a compiler error)
